ethexe/common/src/router.rs

```diff
@@ -0,0 +1,106 @@
+// This file is part of Gear.
+//
+// Copyright (C) 2024 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use alloc::vec::Vec;
+use gear_core::message::ReplyDetails;
+use gprimitives::{ActorId, CodeId, MessageId, H256};
+use parity_scale_codec::{Decode, Encode};
+
+/* Storage related structures */
+
+#[derive(Clone, Debug, Default, Encode, Decode, PartialEq, Eq)]
+pub enum CodeState {
```

Comment:
unused, what is this ?

Reply:
Duplicate type of what's stored in router's map of codes. Will be used once query for code state is implemented